projects/plugins/jetpack/extensions/shared/components/product-management-controls/utils.js

```diff
@@ -0,0 +1,76 @@
+
+/**
+ * Internal dependencies
+ */
+import { PRODUCT_TYPE_PAYMENT_PLAN, PRODUCT_TYPE_SUBSCRIPTION } from './constants';
+
+/**
+ * This list is supposed to be used by the ProductManagementControls component
+ * and the jetpack/membership-products store, based on a given product type.
+ * We've chosen this centralized approach instead of the more common `sprintf`
+ * because it's a bit clearer, generates less cognitive load on the
+ * component's consumers, and it's easier to maintain.
+ *
+ * @see p1648029677784879-slack-CDLH4C1UZ
+ */
+const messages = {
+	'add a new product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Add a new payment plan', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Add a new subscription', 'jetpack' ),
+	},
+	'product not found': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Payment plan not found', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Subscription not found', 'jetpack' ),
+	},
+	'select a product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Select a payment plan', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Select a subscription', 'jetpack' ),
+	},
+	'default new product title': {
+		// The PAYMENT_PLAN message is intentionally the same as SUBSCRIPTION.
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Monthly Subscription', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Monthly Subscription', 'jetpack' ),
+	},
+	'manage your products': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Manage your payment plans.', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Manage your subscriptions.', 'jetpack' ),
+	},
+	'saving product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Saving payment plan…', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Saving subscription…', 'jetpack' ),
+	},
+	'add product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Add payment plan', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Add subscription', 'jetpack' ),
+	},
+	'product requires a name': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Payment plan requires a name', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Subscription requires a name', 'jetpack' ),
+	},
+	'product requires a valid price': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Payment plan requires a valid price', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Subscription requires a valid price', 'jetpack' ),
+	},
+	'successfully created product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __( 'Successfully created payment plan', 'jetpack' ),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __( 'Successfully created subscription', 'jetpack' ),
+	},
+	'there was an error when adding the product': {
+		[ PRODUCT_TYPE_PAYMENT_PLAN ]: __(
+			'There was an error when adding the payment plan.',
+			'jetpack'
+		),
+		[ PRODUCT_TYPE_SUBSCRIPTION ]: __(
+			'There was an error when adding the subscription.',
+			'jetpack'
+		),
+	},
+};
+
+export function getMessageByProductType( message, productType = PRODUCT_TYPE_PAYMENT_PLAN ) {
+	return messages?.[ message ]?.[ productType ] || null;
```

Comment:
This is just personal preference and you can completely ignore this message. For me, it's a bit odd accessing the messages by message and then product type instead of by productType and then message.

messages[productType][message] vs messages[message][productType]

Reply:
I was on the fence between the two implementations.

Yours is easier on the eyes, but I've chosen mine because it keeps similar strings together, which IMHO would simplify translation efforts by virtue of direct comparison.

All in all, I don't have a strong opinion on this, but I'll keep my version to avoid dismissing Jeremy's review 😆